validator/client/beacon-chain-client-factory/beacon_chain_client_factory.go

```diff
@@ -13,7 +14,26 @@ func NewBeaconChainClient(validatorConn validatorHelpers.NodeConnection) iface.B
 	featureFlags := features.Get()
 
 	if featureFlags.EnableBeaconRESTApi {
-		return beaconApi.NewBeaconApiBeaconChainClientWithFallback(validatorConn.GetBeaconApiUrl(), validatorConn.GetBeaconApiTimeout(), grpcClient)
+		return beaconApi.NewBeaconApiBeaconChainClientWithFallback(
+			validatorConn.GetBeaconApiUrl(),
+			validatorConn.GetBeaconApiTimeout(),
+			grpcClient,
+		)
+	} else {
+		return grpcClient
+	}
+}
+
+func NewPrysmBeaconClient(validatorConn validatorHelpers.NodeConnection) iface.PrysmBeaconChainClient {
+	grpcClient := grpcApi.NewGrpcPrysmBeaconChainClient(validatorConn.GetGrpcClientConn())
```

Comment:
No need to initialize this client if the feature flag is set

Reply:
Sure, moved it at the end